rsd-migration-tool/rsd-migration-tool/main.swift

```diff
@@ -0,0 +1,248 @@
+//
+//  main.swift
+//  rsd-migration-tool
+//
+//  Copyright © 2021 Sage Bionetworks. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without modification,
+// are permitted provided that the following conditions are met:
+//
+// 1.  Redistributions of source code must retain the above copyright notice, this
+// list of conditions and the following disclaimer.
+//
+// 2.  Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation and/or
+// other materials provided with the distribution.
+//
+// 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission. No license is granted to the trademarks of
+// the copyright holders even if such marks are included in this software.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+import Foundation
+
+func run() {
```

Comment:
This is incredible.  I’ve never thought about making a migration tool like this, but now that I’ve seen it, I don’t think I can go back, because it’s such a nice courtesy to developers. It would be great to see this sort of thing applied more across the entire software industry. 

I’m excited to try it out but not 100% sure how. Would you mind adding a bit more detail in MigrationNotes_v4.0.md to explain how to run the migration_tool and point it at your project dir?

Reply:
@dephillipsmichael 
Thanks for taking a look at this. I added a little more instruction if you wouldn't mind reading it over. The tool is pretty conservative about what it will automate, and it is a good idea to have and run unit tests after migrating, of course. Also, I am not sure what-all of a current project might *need* to be migrated since the 3.18 version is the same code but it was useful for migrating BridgeApp, MotorControl, and CRF. (Next batch of PRs)